// angular-cc-library/esm5/lib/directives/cvc-format.directive.js
import { __decorate, __param } from "tslib";
import { Directive, ElementRef, HostListener, Optional, Self } from '@angular/core';
import { CreditCard } from '../credit-card';
import { NgControl } from '@angular/forms';
var CvcFormatDirective = /** @class */ (function () {
    function CvcFormatDirective(el, control) {
        this.el = el;
        this.control = control;
        this.target = this.el.nativeElement;
    }
    /**
     * Updates the value to target element, or FormControl if exists.
     * @param value New input value.
     */
    CvcFormatDirective.prototype.updateValue = function (value) {
        if (this.control) {
            this.control.control.setValue(value);
        }
        else {
            this.target.value = value;
        }
    };
    CvcFormatDirective.prototype.onKeypress = function (e) {
        if (!CreditCard.restrictNumeric(e) && !CreditCard.restrictCvc(e.which, this.target)) {
            e.preventDefault();
        }
    };
    CvcFormatDirective.prototype.reformatCvc = function () {
        var _this = this;
        var val = CreditCard.replaceFullWidthChars(this.target.value)
            .replace(/\D/g, '')
            .slice(0, 4);
        var oldVal = this.target.value;
        if (val !== oldVal) {
            this.target.selectionStart = this.target.selectionEnd = CreditCard.safeVal(val, this.target, (function (safeVal) {
                _this.updateValue(safeVal);
            }));
        }
    };
    CvcFormatDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: NgControl, decorators: [{ type: Self }, { type: Optional }] }
    ]; };
    __decorate([
        HostListener('keypress', ['$event'])
    ], CvcFormatDirective.prototype, "onKeypress", null);
    __decorate([
        HostListener('paste'),
        HostListener('change'),
        HostListener('input')
    ], CvcFormatDirective.prototype, "reformatCvc", null);
    CvcFormatDirective = __decorate([
        Directive({
            selector: '[ccCVC]',
        }),
        __param(1, Self()), __param(1, Optional())
    ], CvcFormatDirective);
    return CvcFormatDirective;
}());
export { CvcFormatDirective };
