// ng-maphilight/angular-maphilight.js
import { Component, ElementRef, Input, KeyValueDiffers, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var MaphilightComponent = /** @class */ (function () {
    function MaphilightComponent(elRef, differs) {
        this.elRef = elRef;
        this.el = this.elRef.nativeElement;
        this.configDiffer = differs.find({}).create();
    }
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.img = this.el.querySelector('img');
        this.map = this.el.querySelector('map');
        var /** @type {?} */ id = this.el.getAttribute('id');
        this.img.setAttribute('usemap', '#' + id + ".map");
        this.map.setAttribute('id', id + ".map");
        this.map.setAttribute('name', id + ".map");
        this.drawCanvas();
    };
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.drawCanvas = /**
     * @return {?}
     */
    function () {
        window.jQuery(this.img).maphilight(this.configToApply());
    };
    
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.ngDoCheck = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ changes = this.configDiffer.diff(this.config);
        if (changes) {
            this.drawCanvas();
        }
    };
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.configToApply = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ config = Object.assign({}, this.config);
        if (config.strokeColor && config.strokeColor[0] == '#')
            config.strokeColor = config.strokeColor.substr(1);
        if (config.fillColor && config.fillColor[0] == '#')
            config.fillColor = config.fillColor.substr(1);
        if (config.shadowColor && config.shadowColor[0] == '#')
            config.shadowColor = config.shadowColor.substr(1);
        return config;
    };
    MaphilightComponent.decorators = [
        { type: Component, args: [{
                    selector: 'maphilight',
                    template: "<ng-content> </ng-content> ",
                    styles: [""]
                },] },
    ];
    /** @nocollapse */
    MaphilightComponent.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: KeyValueDiffers, },
    ]; };
    MaphilightComponent.propDecorators = {
        "config": [{ type: Input },],
    };
    return MaphilightComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var MaphilightModule = /** @class */ (function () {
    function MaphilightModule() {
    }
    MaphilightModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [MaphilightComponent],
                    exports: [MaphilightComponent]
                },] },
    ];
    /** @nocollapse */
    MaphilightModule.ctorParameters = function () { return []; };
    return MaphilightModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { MaphilightModule, MaphilightComponent as ɵa };
