// ng-maphilight/__ivy_ngcc__/angular-maphilight.js
import { Component, ElementRef, Input, KeyValueDiffers, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var MaphilightComponent = /** @class */ (function () {
    function MaphilightComponent(elRef, differs) {
        this.elRef = elRef;
        this.el = this.elRef.nativeElement;
        this.configDiffer = differs.find({}).create();
    }
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.img = this.el.querySelector('img');
        this.map = this.el.querySelector('map');
        var /** @type {?} */ id = this.el.getAttribute('id');
        this.img.setAttribute('usemap', '#' + id + ".map");
        this.map.setAttribute('id', id + ".map");
        this.map.setAttribute('name', id + ".map");
        this.drawCanvas();
    };
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.drawCanvas = /**
     * @return {?}
     */
    function () {
        window.jQuery(this.img).maphilight(this.configToApply());
    };
    
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.ngDoCheck = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ changes = this.configDiffer.diff(this.config);
        if (changes) {
            this.drawCanvas();
        }
    };
    /**
     * @return {?}
     */
    MaphilightComponent.prototype.configToApply = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ config = Object.assign({}, this.config);
        if (config.strokeColor && config.strokeColor[0] == '#')
            config.strokeColor = config.strokeColor.substr(1);
        if (config.fillColor && config.fillColor[0] == '#')
            config.fillColor = config.fillColor.substr(1);
        if (config.shadowColor && config.shadowColor[0] == '#')
            config.shadowColor = config.shadowColor.substr(1);
        return config;
    };
    /** @nocollapse */
    MaphilightComponent.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: KeyValueDiffers, },
    ]; };
    MaphilightComponent.propDecorators = {
        "config": [{ type: Input },],
    };
MaphilightComponent.ɵfac = function MaphilightComponent_Factory(t) { return new (t || MaphilightComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.KeyValueDiffers)); };
MaphilightComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MaphilightComponent, selectors: [["maphilight"]], inputs: { config: "config" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function MaphilightComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaphilightComponent, [{
        type: Component,
        args: [{
                selector: 'maphilight',
                template: "<ng-content> </ng-content> ",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.KeyValueDiffers }]; }, { config: [{
            type: Input
        }] }); })();
    return MaphilightComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var MaphilightModule = /** @class */ (function () {
    function MaphilightModule() {
    }
    /** @nocollapse */
    MaphilightModule.ctorParameters = function () { return []; };
MaphilightModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MaphilightModule });
MaphilightModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MaphilightModule_Factory(t) { return new (t || MaphilightModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MaphilightModule, { declarations: function () { return [MaphilightComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [MaphilightComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaphilightModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [MaphilightComponent],
                exports: [MaphilightComponent]
            }]
    }], function () { return []; }, null); })();
    return MaphilightModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { MaphilightModule, MaphilightComponent };
